xcp_d/utils/hcp2fmriprep.py

```diff
@@ -4,165 +4,310 @@
-        lpial,
-        rpial,
-        lwhite,
-        rwhite,
-        linf,
-        rinf,
+    print("finished collecting anat files")
+
+    # Collect functional files to copy
+    subject_task_folders = sorted(
+        glob.glob(os.path.join(anat_dir_orig, "MNINonLinear/Results", "*"))
+    )
+    subject_task_folders = [
+        task for task in subject_task_folders if task.endswith("RL") or task.endswith("LR")
     ]
+    for subject_task_folder in subject_task_folders:
+        # NOTE: What is the first element in the folder name?
+        _, task_id, dir_id = os.path.basename(subject_task_folder).split("_")
+        task_ent = f"task-{task_id}"
+        dir_ent = f"dir-{dir_id}"
+        # TODO: Rename variable
+        filenamex = os.path.basename(subject_task_folder)
+
+        # Find original task files
+        brainmask_orig_temp = os.path.join(subject_task_folder, "brainmask_fs.2.nii.gz")
+
+        bold_nifti_orig = os.path.join(subject_task_folder, f"{filenamex}.nii.gz")
+        bold_nifti_fmriprep = os.path.join(
+            func_dir_fmriprep,
+            f"{sub_ent}_{task_ent}_{dir_ent}_{volspace_ent}_{res_ent}_desc-preproc_bold.nii.gz",
+        )
+        copy_dictionary[bold_nifti_orig] = [bold_nifti_fmriprep]
 
-    for i, j in zip(hcpfiles, newanatfiles):
-        copyfileobj_example(i, j)
-    print("finished converting anat files")
-    # get the task files
+        boldref_orig = os.path.join(subject_task_folder, "SBRef_dc.nii.gz")
+        boldref_fmriprep = os.path.join(
+            func_dir_fmriprep,
+            f"{sub_ent}_{task_ent}_{dir_ent}_{volspace_ent}_{res_ent}_boldref.nii.gz",
+        )
+        copy_dictionary[boldref_orig] = [boldref_fmriprep]
 
-    taskx = glob.glob(hcp_dir + "/Results/*")
-    tasklistx = []
-    print(hcp_dir)
-    for j in taskx:
-        if j.endswith("RL") or j.endswith("LR"):
-            tasklistx.append(j)
+        bold_cifti_orig = os.path.join(
+            subject_task_folder,
+            f"{filenamex}_Atlas_MSMAll.dtseries.nii",
+        )
+        bold_cifti_fmriprep = os.path.join(
+            func_dir_fmriprep,
+            f"{sub_ent}_{task_ent}_{dir_ent}_space-fsLR_den-91k_bold.dtseries.nii",
+        )
+        copy_dictionary[bold_cifti_orig] = [bold_cifti_fmriprep]
+
+        # Grab transforms
+        # TODO: Find actual native-to-T1w transform
+        native_to_t1w_orig = os.path.join(xforms_dir_orig, "acpc_dc2standard.nii.gz")
+        native_to_t1w_fmriprep = os.path.join(
+            func_dir_fmriprep,
+            f"{sub_ent}_{task_ent}_{dir_ent}_from-scanner_to-T1w_mode-image_xfm.nii.gz",
+        )
+        copy_dictionary[native_to_t1w_orig] = [native_to_t1w_fmriprep]
 
-    csf_mask = pkgrf("xcp_d", "/data/masks/csf.nii.gz")
-    wm_mask = pkgrf("xcp_d", "/data/masks/wm.nii.gz")
-    for k in tasklistx:
-        idx = os.path.basename(k).split("_")
-        filenamex = os.path.basename(k)
+        # TODO: Find actual T1w-to-native transform
```

Comment:
Is this the actual T1w-to-native transform, or is this TODO still accurate?

Reply:
still accurate